activerecord/test/cases/inheritance_test.rb

```diff
@@ -208,6 +208,16 @@ def test_new_with_complex_inheritance
     assert_nothing_raised { Client.new(type: 'VerySpecialClient') }
   end
 
+  def test_new_with_store_full_sti_class_option
```

Comment:
The current test name `test_new_with_store_full_sti_class_option` makes me think it uses a `true` value for the option. Can we reword it to reflect that it's disabling the option?